code/modules/projectiles/projectile/special.dm

```diff
@@ -259,19 +259,19 @@
 
 /obj/item/projectile/plasma/on_hit(atom/target)
 	. = ..()
-	if(istype(target, /turf/simulated/mineral))
+	if(ismineralturf(target))

```

Comment:
I like the way TG does this better, personally.

Reply:
How does tg do it?

Reply:
they have a bullet piercing system which we can't really use, but...basically they have X range for travelling in the open air, but they have slight less range when it's piercing mineral turfs.